lib/instance.js

```diff
@@ -322,8 +322,11 @@ Instance.prototype.set = function(key, value, options) { // testhint options:non
           // If attribute is not in model definition, return
           if (!this._isAttribute(key)) {
             if (key.indexOf('.') > -1 && this.Model._isJsonAttribute(key.split('.')[0])) {
-              Dottie.set(this.dataValues, key, value);
-              this.changed(key, true);
+              var previousDottieValue = Dottie.get(this.dataValues, key);
+              if (!_.isEqual(previousDottieValue, value)) {
```

Comment:
Any thoughts on this @mickhansen @janmeier ? I'm happy either way, so if you feel more comfortable with `!==` comparison I will change it.